### DISCUSSION TITLE
auto-merging doesn't show up, even if it is enabled

### DISCUSSION THREAD

**Discussion by odinho** [community/community#50327] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

This feature stopped appearing around the time when the 'Merge queue' was introduced. The option is enabled, but it is no option on our PRs in this same repository.

I guess this is still supposed to work since we see the option, and since there is documentation about [Automatically merging a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request).

![image](https://user-images.githubusercontent.com/1451865/225932773-3dee9638-5490-4928-b4c0-0bea41dc8743.png)

------------

This is a running PR, but there is no auto-merge option here:

![image](https://user-images.githubusercontent.com/1451865/225933153-4a1cbb1f-02b6-44f0-866d-e36ddb21d152.png)

(We tried the Merge queue, but had to disable it since we need to push certain commits directly to the repo from command line for our release process).

**Comment by Anmol-Baranwal** [comment]
The feature should still be available in general, so it's worth checking if there are any specific conditions that need to be met for the feature to appear.

One possible reason could be that the pull request does not meet the merge criteria set by the repository. For example, if the repository has required status checks that must pass before a pull request can be merged, the "Automatically merge pull request" feature may not appear if those checks have not passed.

Another reason could be that there are conflicts in the pull request that need to be resolved before it can be merged. If there are conflicts, the "Automatically merge pull request" feature may not be available until those conflicts are resolved.

**Comment by Mohamed-SayedAlAhl** [comment, marked as answer]
Based on the information provided, it's possible that the "Automatically merge pull request" feature is not available because there are some required status checks that have not passed yet. In the PR description, it says "Some checks haven't completed yet" and "17 in progress checks" which suggests that there are still some checks that are being run. 

It's also worth noting that if the merge queue was enabled previously, it's possible that it could have affected the auto-merge feature. In this case, it's recommended to check if there are any specific settings that need to be adjusted to enable the auto-merge feature again.

Additionally, if the repository has any specific requirements or restrictions set for merging pull requests, such as requiring a specific number of approvals or a certain number of passing status checks, then these requirements must be met before the auto-merge feature can be used.

Finally, it's worth noting that the auto-merge feature may not be available if there are any conflicts between the head and base branches. In this case, the conflicts need to be resolved before the auto-merge feature can be enabled.

**Reply by odinho** [reply]
Indeed, turning on "Merge queue" had removed all required status checks from a part of settings.

Going in and setting some of them to be required again made the 'Auto-merge' feature come back. So just turning it on again, and having status checks is not enough. You have to manually re-enable the required checks it seems like! :)

**Comment by mpstaton** [comment]
Okay GitHub team. I haven't used GitHub actively in some time, and I am about to start coding again. FYI, I am now nearly 45 minutes into trying to find a way to merge my committed branch into main.  

What I seem to be gathering is that it's not possible, I have to do it from command line. (Which, I totally forgot how to do, so that's another hour I would guess of lost time.)  

Or, from this post, there seems to be some sort of review process and then maybe after everything is reviewed it magically shows up as an option?  How do you go through that process?  This kind of information would be helpful.  

I think what people mean when they are raising an issue "There is no merge option in pull request" is "How do I get to the point I can merge?"  There are a few articles I have hit that dance around this issue, and I have yet to see anyone say "Oh, easy, you do steps 1, 2 and 3."

**Reply by mpstaton** [reply]
And, if you don't allow it because complex projects need these kinds of barriers to potential accidents, and it needs to be done on command line, could you also just give me the three lines it takes to do it?  Save me a spirit quest through Stack Overflow and blogs?

And, why would you need this kind of protocol to exist for a project that is initiated by one person and still only has one person?  Seems like overkill.

**Reply by odinho** [reply]
Can you do a screenshot of what you see at the end of your PR?

You first have to create a Pull Request (PR) of your branch, you can go to github.com/<yourorg>/<yourrepo>/pulls and there should be a button there. If you just pushed a change to same branch, it'll helpfully have a yellow box for you to click.

Once you have a PR, you should see the green <merge> button at the end of the page. Do you have that?

In order for you to do *auto merge*. You have to have your tests run automatically using GitHub Actions. And *then* you have to go into your repository settings under "Branch", and edit the "main" branch and add at least one of those actions as required. At that point I believe you'd be given the option to turn on 'auto merge'.

You still need to click the 'auto merge' button for it to go where you want.

If you are only one person, you can also just always commit your branches to main. That's what I often do, then there's no PRs or no extra work. If you did your work on a topic branch you can do it like this:

```bash
git checkout main
git pull # just to make sure
git merge my-topic-branch
git push # this will now push your new main that include newly merged my-topic-branch in it
```

**Comment by pavlo-noumena** [comment]
Same for me, after all pr checks passed, and PR was approved, no auto merge, despite enabled:
![image](https://github.com/community/community/assets/117288860/3678c7aa-88c8-4955-b5ad-6c53d1d8c23f)
![image](https://github.com/community/community/assets/117288860/2953d1ad-2f82-4b20-b3ae-3f590a0ba78a)

Does auto-merge work for anyone at all?

**Reply by odinho** [reply]
Looks like you don't have any required checks? You need to have that first. Or any 'merge requirements' I guess.

**Reply by pavlo-noumena** [reply]
It's clearly stated on the attached screen I have 99 successfully passed checks and  "merging can be performed automatically". 
The skipped check is only triggered on PR title change; that is why it was skipped. I can merge it now by clicking on "Squash and merge", but I need Github to do it automatically if all checks passed successfully.

**Comment by johnnyshankman** [comment]
I have this same issue on all repos where I am admin and owner. The option to auto merge once build checks complete is gone. I believe it's because I'm allowed to override the build checks and immediately merge, but that's silly.

**Reply by jumarko** [reply]
I believe I'm experiencing the problem. I cannot easily try with member permissions but what would be the rationale for admins never seeing this option??

**Comment by mkpmanish** [comment]
Even same thing is happening for me....do we have any fix or work around for this?

**Reply by johnnyshankman** [reply]
None yet.

**Comment by tgross35** [comment]
For everyone who doesn't see this, you need to do the following:

1. Enable auto merge in the general settings
2. Under settings, go to "branches"
3. Add a branch protection rule for your `main` branch if it doesn't exist
4. Within branch protection settings, make sure "Require status checks to pass before merging " is checked
5. Manually add the jobs one by one by searching for them. This is pretty silly, you should just be able to check an "all jobs" box or use wildcards, as mentioned at https://github.com/orgs/community/discussions/12377.
    <img width="765" alt="image" src="https://github.com/community/community/assets/13724985/9d1acbf5-f937-4299-abc3-8edc7a7b86b5">
6. Save

Needing to actually add the individual jobs is the step I always forget, this should really be made more clear in the settings menu (or at least the docs).

**Reply by Magnuti** [reply]
> 1. Enable auto merge in the general settings

Lol completely missed this one. The docs don't even mention it.
